crates/pathfinder/src/rpc/v02/method/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::Context;
+use serde_with::serde_as;
+
+use crate::rpc::serde::{FeeAsHexStr, TransactionVersionAsHexStr};
+use crate::{
+    core::{
+        CallParam, ClassHash, ContractAddress, ContractAddressSalt, Fee, StarknetTransactionHash,
+        TransactionNonce, TransactionSignatureElem, TransactionVersion,
+    },
+    rpc::v02::RpcContext,
+    sequencer::ClientApi,
+};
+
+const fn transaction_version_zero() -> TransactionVersion {
+    TransactionVersion(web3::types::H256::zero())
+}
+
+#[serde_as]
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub struct AddDeployAccountTransactionInput {
+    // Fields from BROADCASTED_TXN_COMMON_PROPERTIES
+    #[serde_as(as = "TransactionVersionAsHexStr")]
+    #[serde(default = "transaction_version_zero")]
+    pub version: TransactionVersion,
+    #[serde_as(as = "FeeAsHexStr")]
+    pub max_fee: Fee,
+    pub signature: Vec<TransactionSignatureElem>,
+    pub nonce: TransactionNonce,
+
+    // Fields from DEPLOY_ACCOUNT_TXN_PROPERTIES
+    pub contract_address_salt: ContractAddressSalt,
+    pub constructor_calldata: Vec<CallParam>,
+    pub class_hash: ClassHash,
```

Comment:
I don't think this is correct. According to the JSON-RPC [specification](https://github.com/starkware-libs/starknet-specs/blob/77de96e5b5465e0fd1dbd40a0e0f20a03b8f3107/api/starknet_write_api.json#L108) the `starknet_addDeployAccount` method takes a single parameter `deploy_account_transaction`, which is of type `BROADCASTED_DEPLOY_ACCOUNT_TXN`. You should add this as a new enum variant of `BroadcastedTransaction` and then simply use that here.

Reply:
Totally missed enclosing `deploy_account_transaction` *shrug*